sktime/forecasting/varmax.py

```diff
@@ -0,0 +1,312 @@
+        error_cov_type = 'unstructured',
+        measurement_error=False,
+        enforce_stationarity=True,
+        enforce_invertibility=True,
+        trend_offset=1,
+        start_params=None,
+        transformed=True,
+        includes_fixed=False,
+        cov_type=None,
+        cov_kwds=None,
+        method='lbfgs',
+        maxiter=50,
+        full_output=1,
+        disp=5,
+        callback=None,
+        return_params=False,
+        optim_score=None,
+        optim_complex_step=None,
+        optim_hessian=None,
+        flags=None,
+        low_memory=False,
+        dynamic=False,
+        information_set='predicted',
+        signal_only=False,
+        random_state=None,
+    ):
+        # Model parameters
+        self.order = order
+        self.trend = trend
+        self.error_cov_type = error_cov_type
+        self.measurement_error = measurement_error
+        self.enforce_stationarity = enforce_stationarity
+        self.enforce_invertibility = enforce_invertibility
+        self.trend_offset = trend_offset
+        self.start_params = start_params
+        self.transformed = transformed
+        self.includes_fixed = includes_fixed
+        self.cov_type = cov_type
+        self.cov_kwds = cov_kwds
+        self.method = method
+        self.maxiter = maxiter
+        self.full_output = full_output
+        self.disp = disp
+        self.callback = callback
+        self.return_params = return_params
+        self.optim_score = optim_score
+        self.optim_complex_step = optim_complex_step
+        self.optim_hessian = optim_hessian
+        self.flags = flags
+        self.low_memory = low_memory
+        self.dynamic = dynamic
+        self.information_set = information_set
+        self.signal_only = signal_only
+
+        super(VARMAX, self).__init__()
+
+    def _fit_forecaster(self, y, X=None):
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : guaranteed to be of a type in self.get_tag("y_inner_mtype")
+            Time series to which to fit the forecaster.
+            Otherwise, if not passed in _fit, guaranteed to be passed in _predict
+        X : optional (default=None)
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : reference to self
+        """
+
+        self._forecaster = _VARMAX(
+            endog=y,
+            exog=X,
+            order = self.order,
+            trend = self.trend,
+            error_cov_type = self.error_cov_type,
+            measurement_error = self.measurement_error,
+            enforce_stationarity = self.enforce_stationarity,
+            enforce_invertibility = self.enforce_invertibility,
+            trend_offset = self.trend_offset,
+        )
+        self._fitted_forecaster = self._forecaster.fit(
```

Comment:
this is sensible, because statsmodels has an interface which is `fit_results = model.fit(data)` and `prediction = model.predict(new_data)`, where `fit_results` is not an estimator type object or a reference to a state updated `model`, but a "fitting results" type objects.

This mirrors R philosophy with stateless container objects, rather than python/sklearn where fitting is state update.

Convention-wise, any fitted parameters should *end* in an underscore, which is not the case here, but let´s not change the adapter for now...

Reply:
Yes, and I think you meant statsmodel has the interface `fit_results = model.fit(data)` and `prediction = fit_results.predict()`?

https://www.statsmodels.org/stable/generated/statsmodels.tsa.statespace.varmax.VARMAXResults.predict.html

Ah okay, noted regarding the convention.